bin/ruby-build

```diff
@@ -1047,6 +1048,16 @@ use_homebrew_yaml() {
   fi
 }
 
+use_homebrew_gmp() {
+  local libdir="$(brew --prefix gmp 2>/dev/null || true)"
+  if [ -d "$libdir" ]; then
+    echo "ruby-build: using gmp from homebrew"
+    package_option ruby configure --with-opt-dir="$libdir"
```

Comment:
You should specify the value of $(brew --prefix) to `--with-opt-dir` instead of libdir.

Reply:
Only `$(brew --prefix)` has side effect for other dependencies. We should keep `--with-opt-dir` values before this function and add gmp path to them.

Reply:
Is there no `--with-gmp-dir`?

Reply:
> Is there no `--with-gmp-dir`?

Yes, It's not yet. We need to use `--with-opt-dir` now.

Reply:
@hsbt Should we add a guard that automatically using `--with-opt-dir` to point to GMP shouldn't happen if the user supplied their own `--with-opt-dir`? Just in case